### PR TITLE
Added an atomic Policy Update functionality

### PR DESCRIPTION
**Pull request by andreygrehov**

### CODE REVIEW

File: manager/sql/manager_sql.go [hunk 1]
```diff
@@ -441,6 +441,22 @@ func (s *SQLManager) Get(id string) (Policy, error) {
...
 }
 
+// Update updates an existing policy.
+func (s *SQLManager) Update(policy Policy) (err error) {
+	tx, err := s.db.Begin()
+	if err != nil {
+		return errors.WithStack(err)
+	}
+
+	if err := s.Delete(policy.GetID()); err != nil {
```
Comment (arekkas) [inline manager/sql/manager_sql.go]: I don't think this works like that, the transaction you're rolling back isn't covering the delete command, so basically, on error, nothing happens. Instead we probably need to create a private function that takes a transaction and executes the delete command, and use that private function in update as well as delete.

**Comment by arekkas** [comment]
Oh and of course, thank you so much for your contribution!

**Comment by arekkas** [comment]
see https://github.com/ory/ladon/pull/75

**Comment by andreygrehov** [comment]
Thanks a lot for taking care of the rest. I didn't get to it.

**Pull request closed without merging**